### PR TITLE
impl(docfx): support `orderedlist` elements

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #10895

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10949)

### CODE REVIEW

File: docfx/doxygen2markdown.cc [hunk 1]
```diff
@@ -662,15 +677,16 @@ bool AppendIfListItem(std::ostream& os, MarkdownContext const& ctx,
                       pugi::xml_node const& node) {
   if (std::string_view{node.name()} != "listitem") return false;
   // The first paragraph is the list item is indented as needed, and starts
-  // with a "- "
+  // with a the item prefix (typically "- " or "1. ").
```
Comment (dbolduc) [inline docfx/doxygen2markdown.cc:680]: s/a the/the/

File: docfx/doxygen2markdown_test.cc [hunk 2]
```diff
@@ -628,6 +657,27 @@ TEST(Doxygen2Markdown, ItemizedListNested) {
   - Sub 3)md");
 }
 
+TEST(Doxygen2Markdown, OrderedListWithParagraphs) {
+  auto constexpr kXml = R"xml(<?xml version="1.0" standalone="yes"?>
+    <doxygen version="1.9.1" xml:lang="en-US">
+        <orderedlist id='test-node'>
+        <listitem><para>Item 1</para><para>More about Item 1</para></listitem>
+        <listitem><para>Item 2: <computeroutput>brrr</computeroutput></para></listitem>
```
Comment (dbolduc) [inline docfx/doxygen2markdown_test.cc:665]: nit: extra indent?

**Review by dbolduc — APPROVED** [review]

File: docfx/doxygen2markdown_test.cc (hunk 2 above)
Comment (coryan) [reply docfx/doxygen2markdown_test.cc]: Done.

File: docfx/doxygen2markdown.cc (hunk 1 above)
Comment (coryan) [reply docfx/doxygen2markdown.cc]: Fixed

**Pull request merged**